ulid.go

```diff
@@ -276,6 +291,55 @@ func (id *ULID) UnmarshalText(v []byte) error {
 	return nil
 }
 
+// UnmarshalTextStrict is like UnmarshalText, but additionally validates that
+// the parsed ULID consists only of valid base32 characters. It is slightly
+// slower than UnmarshalText.
+//
+// ErrDataSize is returned if the len(v) is different from an encoded
+// ULID's length. Invalid encodings return ErrInvalidCharacters.
+func (id *ULID) UnmarshalTextStrict(v []byte) error {
+	// Optimized unrolled loop ahead.
+	if len(v) != EncodedSize {
+		return ErrDataSize
+	}
+
+	// Overflow checking.
+	// See https://github.com/oklog/ulid/issues/9
+	if v[0] > '7' {
+		return ErrOverflow
+	}
+
+	if dec[v[0]] == 0xFF ||
+		dec[v[1]] == 0xFF ||
+		dec[v[2]] == 0xFF ||
+		dec[v[3]] == 0xFF ||
+		dec[v[4]] == 0xFF ||
+		dec[v[5]] == 0xFF ||
+		dec[v[6]] == 0xFF ||
+		dec[v[7]] == 0xFF ||
+		dec[v[8]] == 0xFF ||
+		dec[v[9]] == 0xFF ||
+		dec[v[10]] == 0xFF ||
+		dec[v[11]] == 0xFF ||
+		dec[v[12]] == 0xFF ||
+		dec[v[13]] == 0xFF ||
+		dec[v[14]] == 0xFF ||
+		dec[v[15]] == 0xFF ||
+		dec[v[16]] == 0xFF ||
+		dec[v[17]] == 0xFF ||
+		dec[v[18]] == 0xFF ||
+		dec[v[19]] == 0xFF ||
+		dec[v[20]] == 0xFF ||
+		dec[v[21]] == 0xFF ||
+		dec[v[22]] == 0xFF ||
+		dec[v[23]] == 0xFF ||
+		dec[v[24]] == 0xFF ||
+		dec[v[25]] == 0xFF {
+		return ErrInvalidCharacters
+	}
+	return id.UnmarshalText(v)
```

Comment:
`UnmarshalText` already checks for encoded size as well as overflow. With this implementation we'd be repeating those checks twice.

Reply:
Makes sense. I was originally thinking about preserving the errors returned for length/`v[0]` overflow, but on second thought it doesn't really matter that these new methods return those errors exactly the same.

Reply:
How about factoring out those common checks into a private function that can be reused by both `Parse` and `ParseStrict`?

Reply:
> That's a little tricky because Parse() calls through to UnmarshalText() which also needs to do the length check.

Then we can take out the code from `UnmarshalText` so that it can be reused.

> There's definitely an opportunity for a small optimization here, though. The other thing is that the ErrOverflow check can be omitted after the ErrInvalidCharacters check has been done, since the v[0] > '7' condition is a subset of ErrInvalidCharacters.

Nice.